collect_app/src/test/java/org/odk/collect/android/location/client/GoogleFusedLocationClientTest.java

```diff
@@ -74,30 +76,29 @@ public void startShouldCallLocationClientOnConnected() {
     }
 
     @Test
-    public void stopShouldDisconnectFromGoogleApiIfConnected() {
-
+    public void stopShouldDisconnectFromGoogleApiIfConnected_andAlwaysCallOnClientStopIfListenerSet() {
         TestClientListener testClientListener = new TestClientListener();
         client.setListener(testClientListener);
 
-        // Call through to ApiClient.disconnect(), but don't do anything:
+        // Previously connected, disconnection succeeds
         when(googleApiClient.isConnected()).thenReturn(true);
         client.stop();
+        verify(googleApiClient).disconnect();
+        assertThat(testClientListener.getOnClientStopCount(), is(1));
```

Comment:
Might be easier to use Mockito for this listener verification?

Reply:
Agreed and done. I made all the tests in that class consistent. I didn't touch `AndroidLocationClientTest` which still uses `TestClientListener`.